amr-wind/equation_systems/tke/source_terms/KsgsM84Src.cpp

```diff
@@ -59,13 +61,13 @@ void KsgsM84Src::operator()(
             if (blo.ok()) {
                 amrex::ParallelFor(
                     blo, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                        amrex::Real ceps_local =
-                            (Ceps / 0.93) *
-                            (0.19 + (0.74 * tlscale_arr(i, j, k) / ds));
-                        src_term(i, j, k) += (ceps_local - CepsGround) *
-                                             std::sqrt(tke_arr(i, j, k)) *
-                                             tke_arr(i, j, k) /
-                                             tlscale_arr(i, j, k);
+                        // Remove previous dissipation term
```

Comment:
I don't understand why you remove and add a new dissipation term ...

Reply:
If you look at the original code, that's what it's doing in lines 65-68. The original term which is added globally in lines 49-50 is -ceps_local * sqrt(tke) * tke / tlscale. When the wall model is on, the src term is modified in the wall-adjacent cell by first adding ceps_local * sqrt(tke) * tke / tlscale and then adding -CepsGround * sqrt(tke) * tke / tlscale. That's still what I'm doing in the new implementation, but I'm using the new functions I introduced (calc_dissip) and correcting what is stored as the dissipation term.

Reply:
I agree! 
![Screen Shot 2023-04-21 at 2 24 53 PM](https://user-images.githubusercontent.com/5861743/233728239-30c2624c-80dc-437c-bacf-7d13fe2c55cc.png)